bot/exts/utils/extensions.py

```diff
@@ -151,17 +147,27 @@ def group_extension_statuses(self) -> t.Mapping[str, str]:
 
         return categories
 
-    async def batch_manage(self, action: Action, *extensions: str) -> str:
+    async def batch_manage(self, action: Action, ctx: Context, *extensions: str) -> None:
         """
-        Apply an action to multiple extensions and return a message with the results.
+        Apply an action to multiple extensions, giving feedback to the invoker while doing so.
 
         If only one extension is given, it is deferred to `manage()`.
         """
+        if self.action_in_progress:
```

Comment:
This will block any extensions from loading/unloading while one is going, do we care? We could make the lock unique to each extension.

Reply:
I'm not sure we care, the majority of these will take <5s, so the chance of an accidental overlap by 2 callers is very small.

If a single caller wanted to load.unload multiple exts then they could pass a list to the cog and it would work just fine.

I'm not sure it's worth the complexity.